components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/NamedJdbcTemplate.java

```diff
@@ -0,0 +1,415 @@
+            throw new DataAccessException(JdbcConstants.ErrorCodes.ERROR_CODE_DATABASE_QUERY_PERFORMING_ERROR
+                    .getErrorMessage() + " " + query, e);
+        }
+
+        return result;
+    }
+
+    /**
+     * Executes a query on JDBC and return the result as a domain object.
+     *
+     * @param query            the SQL query with the parameter placeholders.
+     * @param rowMapper        Row mapper functional interface.
+     * @param namedQueryFilter parameters for the SQL query parameter replacement.
+     * @return domain object of required type.
+     */
+    public <T> T fetchSingleRecord(String query, RowMapper<T> rowMapper, NamedQueryFilter namedQueryFilter)
+            throws DataAccessException {
+
+        T result = null;
+        try (Connection connection = dataSource.getConnection()) {
+            NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(connection, query);
+            try (PreparedStatement preparedStatement = namedPreparedStatement.getPreparedStatement()) {
+                if (namedQueryFilter != null) {
+                    namedQueryFilter.filter(namedPreparedStatement);
+                }
+                try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                    if (resultSet.next()) {
+                        result = rowMapper.mapRow(resultSet, 0);
+                    }
+                    if (!resultSet.isClosed() && resultSet.next()) {
+                        logDebug("There are more records than one found for query: {} for the parameters {}", query,
+                                namedQueryFilter);
+                        throw new DataAccessException(JdbcConstants.ErrorCodes.ERROR_CODE_MORE_RECORDS_IN_SINGLE_FETCH.
+                                getErrorMessage() + " " + query);
+                    }
+                }
+            }
+
+        } catch (SQLException e) {
+            logDebug("There has been an error performing the database query. The query is {}, and the parameters" +
+                    " are {}", e, query, rowMapper, namedQueryFilter);
+            throw new DataAccessException(JdbcConstants.ErrorCodes.ERROR_CODE_DATABASE_QUERY_PERFORMING_ERROR
+                    .getErrorMessage() + " " + query, e);
+        }
+
+        return result;
+    }
+
+    /**
+     * Executes the jdbc update query and returns nothing.
+     *
+     * @param query            SQL query with the parameter placeholders.
+     * @param namedQueryFilter parameters for the SQL query parameter replacement.
+     */
+    public void executeUpdate(String query, NamedQueryFilter namedQueryFilter) throws DataAccessException {
+
+        try (Connection connection = dataSource.getConnection()) {
+            NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(connection, query);
+            try (PreparedStatement preparedStatement = namedPreparedStatement.getPreparedStatement()) {
```

Comment:
Can't we use the same getPreparedStatement() method introduced within this class which includes the filter to the statement?

Reply:
Same applies for https://github.com/wso2/carbon-utils/pull/15/files#diff-b121996948ed254f79b9d4ec1d54ddd4R190 as well

Reply:
yes I'll change it accordingly.